src/components/ha-menu-item.ts

```diff
@@ -1,9 +1,12 @@
 import { MdMenuItem } from "@material/web/menu/menu-item";
+import type { CloseMenuEvent } from "@material/web/menu/menu";
 import { css } from "lit";
-import { customElement } from "lit/decorators";
+import { customElement, property } from "lit/decorators";
 
 @customElement("ha-menu-item")
 export class HaMenuItem extends MdMenuItem {
+  @property() closeAction?: (ev: CloseMenuEvent) => void;
```

Comment:
A few suggestions to improve the API:
1.  Turn off the attribute.  I don't know why lit-analyzer doesn't flag that function props shouldn't be attributes.
2. Make the parameter the triggering item instead of the event, and make it optional as many handlers won't need it.
3. To make the purpose much clearer where used in instances, maybe rename to `clickAction` or just `action`.  @bramkragten  any strong opinion here?

```suggestion
  @property({ attribute: false }) clickAction?: (item?: HTMLElement) => void;
```

Reply:
What should be passed as the  `item` here, the `ev.detail.initiator`?

Reply:
Yes